### ISSUE TITLE
Appium Driver, throws null pointer exception in iOS 13 Beta

### ISSUE THREAD

**Issue by chethanshetty09** [appium/appium-desktop#1059]
## Appium or Appium Desktop?

- I have verified that my issue occur with Appium and Appium Desktop as well.

## The problem

While initializing the Appium Driver, throws null pointer exception in iOS 13 Beta. Cannot run my test on Appium 1.13.0 with iOS 13 Beta OS.  Throws: XCApplicationQuery elementSnapshotForDebugDescription]: unrecognized selector sent to instance 0x280e62dc0 error.

With my below code, app will be launched and when it reached to AppDriver, code throws null pointer Exception.

## Environment

java client: 7.0.0
Appium server version : 1.13.0
Desktop OS/version used to run Appium if necessary: 1.13.0
Node.js version : 11.7.0
Mobile platform/version under test: iOS 13 Beta 3
Real device: iPhone 7 plus
MAC version: 10.14.5
Xcode: 11 Beta

```
import io.appium.java_client.AppiumDriver;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.remote.DesiredCapabilities;
import io.appium.java_client.ios.IOSDriver;
import java.net.URL;

public class RunTest {

    public static final String URL = "http://0.0.0.0:4723/wd/hub";

static AppiumDriver driver;
    public static void main(String[] args) throws Exception {

        DesiredCapabilities capabilities = new DesiredCapabilities();
        capabilities.setCapability("platformName", "iOS");
        capabilities.setCapability("deviceName", "iPhone 7 plus");
        capabilities.setCapability("platformVersion", "13.0");
        capabilities.setCapability("udid","5df840e0a62c7adc7c0fe4759fbbbd280a");
        capabilities.setCapability("app", "com.test.demo");
        capabilities.setCapability("browserName", "");
        capabilities.setCapability("deviceOrientation", "portrait");
        capabilities.setCapability("appiumVersion", "1.13.0");

        try {
            new AppiumDriver(new URL(URL), capabilities);
        }catch (Exception e){
            e.printStackTrace();
        }


        driver.quit();
    }
}
```
Exception:
https://gist.github.com/chethanshetty09/3473e33c7515e7f29c0cf392ea2b960f

Appium logs:
https://gist.github.com/chethanshetty09/b0f12f630dd3a8c135d8c80058401010

**Comment by mayureshshirodkar** [comment]
@mykola-mokhnach This seems to be an issue similar to this:
https://github.com/appium/appium/issues/12724

It's been fixed for appium downloaded via npm, but doesnt seem to be fixed for appium desktop.
New WebdriverAgent will be required for this to be fixed, since there are some XCUITest changes that have happened with iOS 12.3.1 and iOS 13

**Comment by KazuCocoa** [comment]
Yes. It works on `appium@beta`.
Appium desktop can connect to local Appium server via custom server tab via _new session window_. Please try via it

**Comment by chethanshetty09** [comment]
Thanks guys i will try and let you know the result.

**Comment by BShivShankar** [comment]
Have tried @KazuCocoa by your suggestion but facing the same error log as i have posted in the link as comment
https://github.com/appium/appium-desktop/issues/1066

Please kindly anyone help on this, :(

**Comment by BShivShankar** [comment]
@chethanshetty09 have you tried, if its working please provide the steps to get the screen to inspect locators in appium desktop
I have tried by pointing to cli appium in custom server appium desktop but no luck

**Comment by chethanshetty09** [comment]
Yes it is not working on iOS 12.3 and above (including iOS 13 Beta). I still see the loading spinner on Appium desktop when try to inspect elements.

**Comment by chethanshetty09** [comment]
Solved here #1066 . Thanks.

**Comment by ravi11296** [comment]
> Yes. It works on `appium@beta`.
> Appium desktop can connect to local Appium server via custom server tab via _new session window_. Please try via it

i have tried it with appium beta and also used appium desktop with both automatic and custom. Still it is spinning into infinite loop.
As i am using xcode 11.3.1 (11C504) iOS 13.3 simulator

getting error as
`Creating session with MJSONWP desired capabilities: {
[BaseDriver]   "NoReset": true,
[BaseDriver]   "bundleId": "xxxx",
[BaseDriver]   "deviceName": "iPad (6th generation)",
[BaseDriver]   "platformName": "iOS",
[BaseDriver]   "platformVersion": "13.3",
[BaseDriver]   "udid": "BD76A030-9E78-45FD-89D1-131997EE9A70",
[BaseDriver]   "newCommandTimeout": 0,
[BaseDriver]   "connectHardwareKeyboard": true
[BaseDriver] }


[WD Proxy] Got an unexpected response with status undefined: {"errno":"ECONNREFUSED","code":"ECONNREFUSED","syscall":"connect","address":"127.0.0.1","port":8100}
[XCUITest] WDA is not listening at 'http://localhost:8100/'
[XCUITest] WDA is currently not running. There is nothing to cache
[XCUITest] Trying to start WebDriverAgent 2 times with 10000ms interval
[XCUITest] These values can be customized by changing wdaStartupRetries/wdaStartupRetryInterval capabilities
[BaseDriver] Event 'wdaStartAttempted' logged at 1581687348018 (19:05:48 GMT+0530 (India Standard Time))
[XCUITest] Launching WebDriverAgent on the device
[WebDriverAgent] Fetching dependencies
[WebDriverAgent] Dependencies up-to-date
[XCUITest] Killing running processes 'xcodebuild.*BD76A030-9E78-45FD-89D1-131997EE9A70, BD76A030-9E78-45FD-89D1-131997EE9A70.*XCTRunner' for the device BD76A030-9E78-45FD-89D1-131997EE9A70...
[XCUITest] 'pgrep -ifn xcodebuild.*BD76A030-9E78-45FD-89D1-131997EE9A70' didn't detect any matching processes. Return code: 1
[XCUITest] 'pgrep -ifn BD76A030-9E78-45FD-89D1-131997EE9A70.*XCTRunner' didn't detect any matching processes. Return code: 1
[XCUITest] Beginning test with command 'xcodebuild build-for-testing test-without-building -project /Applications/Appium.app/Contents/Resources/app/node_modules/appium/node_modules/appium-webdriveragent/WebDriverAgent.xcodeproj -scheme WebDriverAgentRunner -destination id=BD76A030-9E78-45FD-89D1-131997EE9A70 IPHONEOS_DEPLOYMENT_TARGET=13.3 GCC_TREAT_WARNINGS_AS_ERRORS=0 COMPILER_INDEX_STORE_ENABLE=NO' in directory '/Applications/Appium.app/Contents/Resources/app/node_modules/appium/node_modules/appium-webdriveragent'
[XCUITest] Output from xcodebuild will only be logged if any errors are present there. To change this, use 'showXcodeLog' desired capability
[XCUITest] Waiting up to 60000ms for WebDriverAgent to start
[WD Proxy] Matched '/status' to command name 'getStatus'
[WD Proxy] Proxying [GET /status] to [GET http://localhost:8100/status] with no body
[WD Proxy] Got an unexpected response with status undefined: {"errno":"ECONNREFUSED","code":"ECONNREFUSED","syscall":"connect","address":"127.0.0.1","port":8100}
[WD Proxy] Matched '/status' to command name 'getStatus'
[WD Proxy] Proxying [GET /status] to [GET http://localhost:8100/status] with no body
[WD Proxy] Got an unexpected response with status undefined: {"errno":"ECONNREFUSED","code":"ECONNREFUSED","syscall":"connect","address":"127.0.0.1","port":8100}
[XCUITest] Log file for xcodebuild test: /Users/ravikumar/Library/Developer/Xcode/DerivedData/WebDriverAgent-alwvnomvwrdtzoaxbbkniqrpcdpp/Logs/Test/Test-WebDriverAgentRunner-2020.02.14_19-05-50-+0530.xcresult/Staging/2_Test/Diagnostics/WebDriverAgentRunner-E3880B62-3D02-40DB-A56A-F0ECC0A24899/WebDriverAgentRunner-3C4BCE38-DD47-4524-9B55-E2A7A3DB1AF8/Session-WebDriverAgentRunner-2020-02-14_190551-AtLLRi.log
[WD Proxy] Matched '/status' to command name 'getStatus'
[WD Proxy] Proxying [GET /status] to [GET http://localhost:8100/status] with no body
[WD Proxy] Got an unexpected response with status undefined: {"errno":"ECONNREFUSED","code":"ECONNREFUSED","syscall":"connect","address":"127.0.0.1","port":8100}
[WD Proxy] Matched '/status' to command name 'getStatus'
[WD Proxy] Proxying [GET /status] to [GET http://localhost:8100/status] with no body
[WD Proxy] Got an unexpected response with status undefined: {"errno":"ECONNREFUSED","code":"ECONNREFUSED","syscall":"connect","address":"127.0.0.1","port":8100}
[WD Proxy] Matched '/status' to command name 'getStatus'
[WD Proxy] Proxying [GET /status] to [GET http://localhost:8100/status] with no body`
